### PR TITLE
ci: update release management workflow from semantic-release to changesets

### PR DESCRIPTION
**Pull request by thlmenezes**
## Description of changes

- Replace [`semantic-release`](https://github.com/semantic-release/semantic-release#readme) release workflow with [`changesets`](https://github.com/changesets/changesets#readme)

### Current workflow

1. Every PR merged triggers a new release
1. [Release action](https://github.com/devopness/devopness/actions/workflows/sdk-javascript-release.yml):
    1. Creates a new tag to track package version
    1. Publishes a new package version to npm
    1. Creates a new release with the description including a changelog extracted from commit messages

> "Following formalized conventions for commit messages, semantic-release automatically determines the next semantic version number, generates a changelog and publishes the release."
> Source: https://semantic-release.gitbook.io/semantic-release#how-does-it-work

#### Limitations

> [!IMPORTANT]
> These are the limitations of `devopness/devopness` current release management workflow, this list may includes topics that have no relation with `semantic-release` package

1. Versioning multiple packages in the same repo has no official support

> "there are no short term plans to support monorepos natively in semantic-release. any attempt to support them would need to come from the community for the foreseeable future."
> Source: https://github.com/semantic-release/semantic-release/issues/193#issuecomment-1590410244

2. Contributors are limited to PR title content to include changelog entries, no further content can be added
1. Changelogs do not mention contributors
1. Changelogs can only be accessed using GitHub Release, no `CHANGELOG.md` file
1. Every PR triggers a new release: if you approve a PR, it is already shipping; no review, no second chances

### Proposed workflow

1. A new PR merged may, or may not, include `changesets`
    1.  [What is a changeset?](https://github.com/changesets/changesets/blob/main/docs/adding-a-changeset.md#what-is-a-changeset)
1. If a changeset is detected, a new PR "Version Packages" is opened by release action, this PR will:
    1. update CHANGELOG `/packages/.../CHANGELOG.md`
    1. update related package versions `/packages/.../package.json#version`
    1. delete changesets
1. If a package version was updated locally but not on npm, `Release action` will :
    1. Publish new package version to npm
    1. Create new tag namespaced by package, i.e "@devopness/sdk-js@2.147.0"
    1. Create new release on repo, mentioning contributors

#### Does it solve previous limitations?

TL;DR yes

> Versioning multiple packages in the same repo has no official support

Changesets project is focused on multi package repositories, and there are [many examples on GitHub](https://github.com/search?utf8=%E2%9C%93&q=path%3A.changeset&type=code)

> Contributors are limited to PR title content to include changelog entries, no further content can be added

Contributors may write a changeset, describing

> - What we need to release
> - What version we are releasing packages at (using a [semver bump type](https://semver.org/))
> - A changelog entry for the released packages
>
> Source: [Adding a changeset - What is a changeset?](https://github.com/changesets/changesets/blob/main/docs/adding-a-changeset.md#what-is-a-changeset)

This way when making the change, the contributor also documents the benefits of the solution

> Changelogs do not mention contributors

Changelogs will mention contributors, e.g https://github.com/Shopify/polaris/releases/tag/%40shopify%2Fpolaris%4013.9.1

> Changelogs can only be accessed using GitHub Release, no `CHANGELOG.md` file

A CHANGELOG.md file will be stored on the repo

> Every PR triggers a new release: if you approve a PR, in the next second is already shipping; no review, no second chances

New releases and changelogs will be reviewed in a separate PR from code contribution.

## GitHub issues resolved by this PR

N/A

## Quality Assurance

- Once the changes in this PR are merged and deployed, success criteria is: Proposed release management workflow can be implemented on repo

## More info

[DVN-16680]

[DVN-16680]: https://devopness.atlassian.net/browse/DVN-16680?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -0,0 +1,39 @@
+name: Release Packages
+
+on:
+  workflow_dispatch:
+  push:
+    paths:
+      - packages/sdks/javascript/src/**
+      - packages/sdks/javascript/package.json
+      - packages/ui/react/src/**
+      - packages/ui/react/package.json
```
Comment (jfoliveira) [inline .github/workflows/release.yml:10]: Perhaps use wildcards so we don't need to update this file every time a new package is added to the repo?

```suggestion
      - packages/**/src/**
      - packages/**/package.json
```

**Review by jfoliveira — CHANGES_REQUESTED** [review]

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -0,0 +1,37 @@
+name: Release Packages
```
Comment (jfoliveira) [inline .github/workflows/release.yml:1]: @thlmenezes could you please fix linter errors before merging this PR?
https://github.com/devopness/devopness/actions/runs/11633880424/job/32399935648?pr=1003#step:6:13 

image

**Review by jfoliveira — CHANGES_REQUESTED** [review]

File: .github/workflows/release.yml (hunk 2 above)
Comment (thlmenezes) [reply .github/workflows/release.yml:1]: Thanks for the review, was not expecting any lint errors without any code changes on the package files that did not happen before 👀 

Apparently was due to some missing dependencies and configurations.

1. Issues resolved;
1. Updated ESLint configuration to use `import` plugin recommended defaults;
1. Updated lint command to fail on warnings as well, allowing for stricter CI validations.

Could you review this PR again?

**Review by jfoliveira — APPROVED** [review]

**Review by Diegiwg — APPROVED** [review]
